api/v1alpha1/common_validation.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/robfig/cron/v3"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+const (
+	// ValidateSchedulerError defined the error message for ValidateScheduler
+	ValidateSchedulerError = "schedule and duration should be omitted or defined at the same time"
+
+	// ValidatePodchaosSchedulerError defined the error message for ValidateScheduler of Podchaos
+	ValidatePodchaosSchedulerError = "schedule should be omitted"
+
+	// ValidateValueParseError defined the error message for value parse error
+	ValidateValueParseError = "value parse error"
+)
+
+// ValidateScheduler validates the scheduler and duration
+func ValidateScheduler(duration *string, scheduler *SchedulerSpec, spec *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	schedulerField := spec.Child("scheduler")
+	durationField := spec.Child("duration")
+
+	if duration != nil && scheduler != nil {
+		_, err := cron.ParseStandard(scheduler.Cron)
```

Comment:
For #373 you can do type check on the cron, if the syntax is `every 5m` then the type should be  `ConstantDelaySchedule`. However for normal cron syntax it's a bit hard to compare, you need to check the unit.

Reply:
@at15 
It is better to solve #373  on another PR. Otherwise current PR will get complicated. 😂